frontend/javascripts/oxalis/view/right-border-tabs/starting_job_modals.tsx

```diff
@@ -297,7 +303,19 @@ function StartingJobModal(props: StartingJobModalProps) {
     initialOutputSegmentationLayerName || "segmentation"
   }_corrected`;
   // TODO: Other jobs also have an output segmentation layer. The names for these jobs should also be configurable.
-  const hasOutputSegmentationLayer = jobName === "materializing this volume annotation";
+  const hasOutputSegmentationLayer = jobName === JobNames.MATERIALIZE_VOLUME_ANNOTATION;
+  const notAllowedOutputLayerNames = allLayer
+    .filter((layer) => {
+      // Filtering out the layer that is currently selected as this name is allowed
+      // because the output layer overwrites the selected layer.
```

Comment:
```suggestion
      // Existing layer names may not be used for the output layer. The only exception is the name of the currently selected layer.
      // If that layer name is chosen, the original layer data won't exist in the output dataset, since it will be replaced by the new output.
```

Maybe like this? Needs prettier-formatting, though. You could also keep the comment as is. My interpretation didn't really make sense, which is why I'd hope I won't question the veracity of the comment in the future :laughing:

Reply:
> // If that layer name is chosen, the original layer data won't exist in the output dataset, since it will be replaced by the new output.

Technically speaking, this is not correct. The selected layer will never directly exist in the output dataset, if the job has an output layer. Therefore, this is even the case if the layer name is not chosen. But your 3rd sentence reads like it is only the case when the layer name is chosen.

I rephrased your suggestion a little and made it more expressive:

       // is the name of the currently selected layer. This layer is the only one not
       // copied over from the original dataset to the output dataset.
       // Therefore, this name is available as the name for the output layer name.
       // That is why that layer is filtered out here.

Reply:
Makes sense :+1: